drivers/sensor/opt3001/opt3001.c

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2019 Actinius
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <i2c.h>
+#include <sensor.h>
+#include <logging/log.h>
+#include <misc/__assert.h>
+
+#include "opt3001.h"
+
+LOG_MODULE_REGISTER(opt3001, CONFIG_SENSOR_LOG_LEVEL);
+
+static int opt3001_reg_read(struct opt3001_data *drv_data, u8_t reg,
+			    u16_t *val, bool send_stop)
+{
+	u8_t value[2];
+
+	struct i2c_msg msgs[2] = {
+		{
+			.buf = &reg,
+			.len = 1,
+			.flags = I2C_MSG_WRITE,
+		},
+		{
+			.buf = value,
+			.len = 2,
+			.flags = I2C_MSG_READ,
+		},
+	};
+
+	if (send_stop) {
+		msgs[1].flags |= I2C_MSG_STOP;
+	}
+
+	if (i2c_transfer(drv_data->i2c, msgs,
```

Comment:
`i2c_burst_read` would be better here, as you would not have to manually construct the msgs array

Reply:
Indeed. Updated.